tests/adam_test.cpp

```diff
@@ -418,3 +418,51 @@ TEST_CASE("PadamLogisticRegressionTest", "[AdamTest]")
       coordinates);
   REQUIRE(testAcc == Approx(100.0).epsilon(0.006)); // 0.6% error tolerance.
 }
+
+TEST_CASE("SimpleAdamWTestFunction", "[AdamTest]")
+{
+  // Sometimes this test can fail randomly, so we allow it to run up to three
+  // times.
+  bool success = false;
+  for (size_t trial = 0; trial < 3; ++trial)
+  {
+    SGDTestFunction f;
+    AdamW optimizer(1e-2, 1, 0.9, 0.99, 0.25, 1e-8);
```

Comment:
Do you have some insight how often the test fails? Maybe we should tweak the optimizer parameter?

Reply:
Sorry about that , it seems that it used to failed because I parameterised it wrong. I will fix it.